src/ray/raylet/node_manager.cc

```diff
@@ -344,6 +344,68 @@ void NodeManager::ProcessClientMessage(std::shared_ptr<LocalClientConnection> cl
     ObjectID object_id = from_flatbuf(*message->object_id());
     RAY_LOG(DEBUG) << "reconstructing object " << object_id;
     RAY_CHECK_OK(object_manager_.Pull(object_id));
+
+    // If the blocked client is a worker, then release any CPU resources that
+    // it acquired for its assigned task while it is blocked. The resources
+    // will be acquired again once the worker is unblocked.
+    std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
+    if (worker) {
```

Comment:
you can combine these two if statements into one `if (worker && !worker->IsBlocked())`. I personally would take it a step further and use the opposite logic with early exit : `if (!worker || worker->IsBlocked()) break;`

Reply:
I prefer not to use `break;` statements whenever possible, since it is usually less error-prone if you return in only one place.

Reply:
I always prefer structuring my code with early termination logic. That way you don't have to scroll through code searching for any `else`s or any other code paths executed for all cases. Early termination logic is easier to follow. It also saves on indentation and is more assembly-like.